File: clients/typescript/examples/collections-date.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const collection = database.collection("**COLLECTION_NAME**");

(async function () {
  // Use dates in insertions
  await collection.insertOne({ dateOfBirth: new Date(1394104654000) });
  await collection.insertOne({ dateOfBirth: new Date("1863-05-28") });

  // Use $currentDate in an update
  await collection.updateOne(
    {
      dateOfBirth: new Date("1863-05-28"),
    },
    {
      $set: { message: "Happy Birthday!" },
      $currentDate: { lastModified: true },
    },
  );

  // Use a date in a filter
  const found = await collection.findOne({
    dateOfBirth: { $lt: new Date("1900-01-01") },
  });
})();

File: clients/python/examples/collections-date.py
from datetime import datetime

from astrapy import DataAPIClient
from astrapy.data_types import DataAPITimestamp

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Use DataAPITimestamp in insertions
collection.insert_one(
    {"when": DataAPITimestamp.from_string("2024-12-06T12:34:56.789Z")}
)
collection.insert_one(
    {"registered_at": DataAPITimestamp.from_datetime(datetime.now())}
)

# Use $current date in an update
collection.update_one(
    {"registered_at": {"$exists": True}},
    {"$currentDate": {"last_reviewed": True}},
)

# Use DataAPITimestamp in an equality filter
collection.update_one(
    {
        "registered_at": DataAPITimestamp.from_string(
            "2024-12-06T12:34:56.789Z"
        )
    },
    {"$set": {"message": "happy Sunday!"}},
)

# Use DataAPITimestamp in a "less than" filter
collection.find_one(
    {
        "registered_at": {
            "$lt": DataAPITimestamp.from_string(
                "2025-12-06T12:34:56.789Z"
            )
        }
    }
)

# Print a document with a date
# Will print something like:
# {'registered_at': DataAPITimestamp(timestamp_ms=1733488496789 [2024-12-06T12:34:56.789Z])}
print(
    collection.find_one(
        {
            "registered_at": {
                "$lt": DataAPITimestamp.from_string(
                    "2025-12-06T12:34:56.789Z"
                )
            }
        },
        projection={"_id": False},
    )
)
